src/app/themes.tsx

```diff
@@ -10,53 +10,121 @@ import {
     createTheme as createMaterialTheme,
 } from "@mui/material";
 
-type CustomTheme = {
-    foregroundColor: string;
-    backgroundColor: string;
-    primaryForegroundColor: string;
-    primaryBackgroundColor: string;
-    secondaryForegroundColor: string;
-    secondaryBackgroundColor: string;
-    accentBackgroundColor: string;
-    accentForegroundColor: string;
-    surfaceForegroundColor: string;
-    surfaceBackgroundColor: string;
-    errorColor: string;
+export type standardPalette = {
+    background: string[];
+    foreground: string[];
+    largeForeground: string[];
 };
 
-const lightTheme: CustomTheme = {
-    foregroundColor: "#000000",
-    backgroundColor: "#eeeeee",
-    primaryForegroundColor: "#000000",
-    primaryBackgroundColor: "#63a036",
-    secondaryForegroundColor: "#ffffff",
-    secondaryBackgroundColor: "#066940",
-    accentBackgroundColor: "#609fd3",
-    accentForegroundColor: "#000000",
-    surfaceForegroundColor: "#000000",
-    surfaceBackgroundColor: "#ffffff",
-    errorColor: "#ff624e",
+export type rainbowPalette = {
+    red: standardPalette;
+    orange: standardPalette;
+    yellow: standardPalette;
+    lightGreen: standardPalette;
+    darkGreen: standardPalette;
+    blue: standardPalette;
+    purple: standardPalette;
+    grey: standardPalette;
+    brown: standardPalette;
 };
 
-const darkTheme: CustomTheme = {
-    foregroundColor: "#ffffff",
-    backgroundColor: "#1a1a1a",
-    primaryForegroundColor: "#ffffff",
-    primaryBackgroundColor: "#066940",
-    secondaryForegroundColor: "#000000",
-    secondaryBackgroundColor: "#63a036",
-    accentBackgroundColor: "#1b629c",
-    accentForegroundColor: "#ffffff",
-    surfaceForegroundColor: "#ffffff",
-    surfaceBackgroundColor: "#161414",
-    errorColor: "#ff624e",
+export type customTheme = {
+    light: boolean;
+    main: standardPalette & { border: string };
+    primary: standardPalette;
+    error: string;
+    shadow: string;
+    rainbow: rainbowPalette;
+};
+
+export const lightTheme: customTheme = {
+    light: true,
+    main: {
+        background: ["#fdfdfd", "#eeeeee", "#eeeeee", "#eeeeee"],
```

Comment:
Why do we have `#eeeeee` 3 times?

Reply:
The text in front is different color - it is just mainly for users who use to match e.g. foreground[2] with background[2] as they are checked to have passed the accessibility test already - essentially it means if you use lighter foreground color just stick with the same background color 

The first two [0] and [1] are mainly for setting different background color, [2] and [3] are just mainly for giving greyer text which is used in e.g. login panel